irohad/ametsuchi/impl/rocksdb_storage_impl.cpp

```diff
@@ -0,0 +1,228 @@
+#include "ametsuchi/impl/mutable_storage_impl.hpp"
+#include "ametsuchi/impl/peer_query_wsv.hpp"
+#include "ametsuchi/impl/rocksdb_block_query.hpp"
+#include "ametsuchi/impl/rocksdb_command_executor.hpp"
+#include "ametsuchi/impl/rocksdb_common.hpp"
+#include "ametsuchi/impl/rocksdb_indexer.hpp"
+#include "ametsuchi/impl/rocksdb_query_executor.hpp"
+#include "ametsuchi/impl/rocksdb_settings_query.hpp"
+#include "ametsuchi/impl/rocksdb_specific_query_executor.hpp"
+#include "ametsuchi/impl/rocksdb_temporary_wsv_impl.hpp"
+#include "ametsuchi/impl/rocksdb_wsv_command.hpp"
+#include "ametsuchi/impl/rocksdb_wsv_query.hpp"
+#include "ametsuchi/impl/temporary_wsv_impl.hpp"
+#include "ametsuchi/ledger_state.hpp"
+#include "ametsuchi/tx_executor.hpp"
+#include "common/bind.hpp"
+#include "common/result.hpp"
+#include "logger/logger.hpp"
+#include "logger/logger_manager.hpp"
+
+namespace iroha::ametsuchi {
+
+  RocksDbStorageImpl::RocksDbStorageImpl(
+      std::shared_ptr<RocksDBContext> db_context,
+      boost::optional<std::shared_ptr<const iroha::LedgerState>> ledger_state,
+      std::shared_ptr<BlockStorage> block_store,
+      std::shared_ptr<shared_model::interface::PermissionToString>
+          perm_converter,
+      std::shared_ptr<PendingTransactionStorage> pending_txs_storage,
+      std::shared_ptr<shared_model::interface::QueryResponseFactory>
+          query_response_factory,
+      std::unique_ptr<BlockStorageFactory> temporary_block_storage_factory,
+      std::optional<std::reference_wrapper<const VmCaller>> vm_caller,
+      logger::LoggerManagerTreePtr log_manager)
+      : StorageBase(std::move(ledger_state),
+                    std::move(block_store),
+                    std::move(perm_converter),
+                    std::move(pending_txs_storage),
+                    std::move(query_response_factory),
+                    std::move(temporary_block_storage_factory),
+                    std::move(vm_caller),
+                    std::move(log_manager),
+                    "prepared_block_",
+                    false),
+        db_context_(std::move(db_context)) {}
+
+  std::unique_ptr<TemporaryWsv> RocksDbStorageImpl::createTemporaryWsv(
+      std::shared_ptr<CommandExecutor> command_executor) {
+    auto rdb_command_executor =
+        std::dynamic_pointer_cast<RocksDbCommandExecutor>(command_executor);
+    if (rdb_command_executor == nullptr) {
+      throw std::runtime_error("Bad CommandExecutor cast!");
+    }
+    // if we create temporary storage, then we intend to validate a new
+    // proposal. this means that any state prepared before that moment is
+    // not needed and must be removed to prevent locking
+    command_executor->skipChanges();
+    return std::make_unique<RocksDbTemporaryWsvImpl>(
+        std::move(rdb_command_executor),
+        logManager()->getChild("TemporaryWorldStateView"));
+  }
+
+  iroha::expected::Result<std::unique_ptr<QueryExecutor>, std::string>
```

Comment:
[minor] could be just `expected::Result<std::unique_ptr<QueryExecutor>, std::string>`